log/loggers.go

```diff
@@ -5,248 +5,255 @@ import (
 	"log"
 )
 
-// Info takes a pointer subLogger struct and string sends to StageLogEvent
-func Info(sl *SubLogger, data string) {
+// Infoln is a logging function that takes a sublogger and an arbitrary number
+// of interface{} arguments. This writes to configured io.Writer(s) as an
+// information message using default formats for its operands. A new line is
+// automatically added to the output.
+func Infoln(sl *SubLogger, a ...interface{}) {
 	mu.RLock()
 	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil {
-		return
-	}
-	if fields.info {
-		fields.output.StageLogEvent(func() string { return data },
-			fields.logger.InfoHeader,
-			fields.name,
-			fields.logger.Spacer,
-			fields.logger.TimestampFormat,
-			fields.logger.ShowLogSystemName,
-			fields.logger.BypassJobChannelFilledWarning)
-	}
-	logFieldsPool.Put(fields)
+	fields.stageln(fields.logger.InfoHeader, a...)
 }
 
-// Infoln takes a pointer subLogger struct and interface sends to StageLogEvent
-func Infoln(sl *SubLogger, v ...interface{}) {
+// InfolnWithFields is a logging function that takes a sublogger, additional
+// structured logging fields and an arbitrary number of interface{} arguments.
+// This writes to configured io.Writer(s) as an information message using
+// default formats for its operands. A new line is automatically added to the
+// output. If structured logging is not enabled, the fields will be ignored.
+func InfolnWithFields(sl *SubLogger, extra ExtraFields, a ...interface{}) {
 	mu.RLock()
 	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil {
-		return
-	}
-	if fields.info {
-		fields.output.StageLogEvent(func() string { return fmt.Sprintln(v...) },
-			fields.logger.InfoHeader,
-			fields.name,
-			fields.logger.Spacer,
-			fields.logger.TimestampFormat,
-			fields.logger.ShowLogSystemName,
-			fields.logger.BypassJobChannelFilledWarning)
-	}
-	logFieldsPool.Put(fields)
+	fields.structuredFields = extra
+	fields.stageln(fields.logger.InfoHeader, a...)
 }
 
-// Infof takes a pointer subLogger struct, string and interface formats sends to StageLogEvent
-func Infof(sl *SubLogger, data string, v ...interface{}) {
+// Infof is a logging function that takes a sublogger, a format string along
+// with optional arguments. This writes to configured io.Writer(s) as an
+// information message which formats according to the format specifier.
+// A new line is automatically added to the output.
+func Infof(sl *SubLogger, format string, a ...interface{}) {
 	mu.RLock()
 	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil {
```

Comment:
The removal of this is causing problems because `globalLogConfig` is not set in tests, so it always returns nil and always panics if a log is referenced in any executed code